### PR TITLE
Console: Alerts refactoring

### PR DESCRIPTION
**Pull request by vrockai**
The code is less fragmented, is using  shared services/directives/controllers and is much more clear in general. + contains several bug fixes.

### CODE REVIEW

File: console/src/main/scripts/plugins/directives/alert/html/alert.html [hunk 1]
```diff
@@ -1,54 +1,47 @@
...
-      <strong>Response time</strong> was above the threshold ({{alert.threshold | number: 0}} ms) for
 
-      <hk-time-interval hk-time="alert.durationTime"></hk-time-interval>
-
-      (until
-      <span ng-show="alert.isThisYear && alert.isToday">{{alert.end | date:'HH:mm' }}).</span>
-      <span ng-show="alert.isThisYear && !alert.isToday">{{alert.end | date:'d MMM HH:mm:ss' }}).</span>
-      <span ng-show="!alert.isThisYear">{{alert.end | date:'d MMM y HH:mm:ss' }}).</span>
-      The average response time was {{alert.avg  | number: 0}} ms.
+    <div class="hk-event" ng-switch on="alert.alertType">
```
Comment (mtho11) [inline console/src/main/scripts/plugins/directives/alert/html/alert.html]: Is the 'on' attribute valid or necessary with ng-switch?

Comment (vrockai) [inline console/src/main/scripts/plugins/directives/alert/html/alert.html]: valid but not necessary

File: console/src/main/scripts/plugins/directives/components/html/hk-switch.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <input type="checkbox"
+         class="onoffswitch-checkbox"
+         ng-model="hkModel"
+         ng-disabled="hkDisabled"
+         ng-click="hkClick()"
+         ng-change="hkChange()"
+         id="{{id}}">
+  <label class="onoffswitch-label" for="{{id}}">
+    <span class="onoffswitch-inner">
+      <span class="onoffswitch-active ng-binding">{{hkOnText}}</span>
```
Comment (mtho11) [inline console/src/main/scripts/plugins/directives/components/html/hk-switch.html]: What is ng-binding ?  I know ng-bind.

**Comment by mtho11** [comment]
I have made my comments. Merging since there are many changes that might conflict with what I'm working on.

File: console/src/main/scripts/plugins/directives/components/html/hk-switch.html (hunk 2 above)
Comment (vrockai) [inline console/src/main/scripts/plugins/directives/components/html/hk-switch.html]: copy-paste error, should not be there.

**Pull request merged**